firefox-ios/Client/Frontend/Browser/TermsOfUse/TermsOfUseViewController.swift

```diff
@@ -0,0 +1,309 @@
+        super.traitCollectionDidChange(previousTraitCollection)
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass {
+            setupConstraints()
+            view.layoutIfNeeded()
+        }
+    }
+
+    private func setupUI() {
+        view.backgroundColor = UIColor.black.withAlphaComponent(UX.backgroundAlpha)
+        view.addSubview(sheetContainer)
+        sheetContainer.addSubview(grabberView)
+        sheetContainer.addSubview(stackView)
+        setupConstraints()
+        setupDismissGesture()
+        setupPanGesture()
+        addStackSubviews()
+    }
+
+    private func setupConstraints() {
+        NSLayoutConstraint.deactivate(activeContainerConstraints)
+
+        var containerConstraints = [
+            sheetContainer.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            sheetContainer.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ]
+
+        if traitCollection.horizontalSizeClass == .regular {
+            containerConstraints.append(contentsOf: [
+                sheetContainer.widthAnchor.constraint(equalToConstant: UIScreen.main.bounds.width * UX.iPadWidthMultiplier),
+                sheetContainer.leadingAnchor.constraint(greaterThanOrEqualTo: view.leadingAnchor,
+                                                        constant: UX.sheetContainerSidePadding),
+                sheetContainer.trailingAnchor.constraint(lessThanOrEqualTo: view.trailingAnchor,
+                                                         constant: -UX.sheetContainerSidePadding)
+            ])
+        } else {
+            containerConstraints.append(contentsOf: [
+                sheetContainer.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                sheetContainer.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+            ])
+        }
+
+        NSLayoutConstraint.activate(containerConstraints)
+        activeContainerConstraints = containerConstraints
+
+        NSLayoutConstraint.activate([
+            grabberView.topAnchor.constraint(equalTo: sheetContainer.topAnchor, constant: UX.grabberTopPadding),
+            grabberView.centerXAnchor.constraint(equalTo: sheetContainer.centerXAnchor),
+            grabberView.widthAnchor.constraint(equalToConstant: UX.grabberWidth),
+            grabberView.heightAnchor.constraint(equalToConstant: UX.grabberHeight),
+
+            stackView.leadingAnchor.constraint(equalTo: sheetContainer.leadingAnchor, constant: UX.stackSidePadding),
+            stackView.trailingAnchor.constraint(equalTo: sheetContainer.trailingAnchor, constant: -UX.stackSidePadding),
+            stackView.topAnchor.constraint(equalTo: grabberView.bottomAnchor, constant: UX.stackSpacing),
+            stackView.bottomAnchor.constraint(equalTo: sheetContainer.bottomAnchor, constant: -UX.stackSidePadding)
+        ])
+    }
+
+    private func addStackSubviews() {
+        stackView.addArrangedSubview(logoImageView)
```

Comment:
nit: By the way, do you think we could add this functionality to the stack view’s initializer?

Reply:
yes, i added this change. Thank you!

Reply:
I actually prefer it outside the initializers since we have a couple of places where we use `setupViews` and we can see all the stack arrangements in one place.

Reply:
ok, I moved it outside the initializers